src/java.base/share/classes/jdk/internal/util/Architecture.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2018, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.util;
+
+import jdk.internal.vm.annotation.ForceInline;
+
+/**
+ * System architecture enum values.
+ * Each architecture has a matching {@code public static boolean isXXX()} method
+ * that is true when running on that architecture.
+ * The enum values correspond to architectures as named in the build system.
+ * See the values of `OPENJDK_TARGET_ARCH_OSARCH`.
+ * The order and ordinal values must match the values defined in OperatingSystemProps.java.template.
+ */
+public enum Architecture {
+    X64(),        // Represents AMD64 and X86_64
+    X86(),
+    IA64(),
+    ARM(),
+    AARCH64(),
+    RISCV64(),
+    S390X(),
+    PPC64LE(),
+    ;
+
+    // Cache a copy of the array for lightweight indexing
+    private static final Architecture[] archValues = Architecture.values();
+
+    /**
+     * {@return {@code true} if the current architecture is X64, Aka amd64}
+     */
+    @ForceInline
+    public static boolean isX64() {
+        return OperatingSystemProps.TARGET_ARCH_IS_X64;
+    }
+
+    /**
+     * {@return {@code true} if the current architecture is X86}
+     */
+    @ForceInline
+    public static boolean isX86() {
+        return OperatingSystemProps.TARGET_ARCH_IS_X86;
+    }
+
+    /**
+     * {@return {@code true} if the current architecture is IA64}
+     */
+    @ForceInline
+    public static boolean isIA64() {
+        return OperatingSystemProps.TARGET_ARCH_IS_IA64;
+    }
+
+    /**
+     * {@return {@code true} if the current architecture is Arm}
+     */
+    @ForceInline
+    public static boolean isARM() {
```

Comment:
It should define what’s the difference to aarch64 for example will aarch64 also be arm, but arm32 wont? (Or remove)

Reply:
> It should define what’s the difference to aarch64 for example will aarch64 also be arm, but arm32 wont? (Or remove)

I think x86 and ARM are a bit confusing in this regard, as they can refer to 32-bit architectures in a narrow sense and 32-bit or 64 bit architectures in a broad sense.

For clarity, we can use more specific names: `x86-32`/`i686` and `arm32`/`aarch32`. But these names don't seem to be that commonly used.

Reply:
```suggestion
    public static boolean isARM32() {
```